system/lib/wasmfs/syscalls.cpp

```diff
@@ -271,7 +263,11 @@ long __syscall_fstat64(long fd, long buf) {
   } else if (file->is<DataFile>()) {
     buffer->st_mode |= S_IFREG;
   }
-  buffer->st_ino = fd;
+  buffer->st_ino = (ino_t)file.get();
+  // Set inode number to the file pointer. This gives a unique inode number for
+  // each file in the simplest possible way.
+  // TODO: For security it would probably be better to use an indirect mapping.
+  buffer->st_ino = (ino_t)file.get();
```

Comment:
How about adding a static assert here that a pointer can fit in `ino_t`? The dangerous case is in wasm64 pointers are 64-bits, and then if `ino_t` is 32-bit then we could get collisions after truncation.

Reply:
Yep, added to the previous PR, but I can bring it in here too.